Modules/_abc.c

```diff
@@ -88,7 +81,8 @@ abc_data_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
     self->_abc_registry = NULL;
     self->_abc_cache = NULL;
     self->_abc_negative_cache = NULL;
-    self->_abc_negative_cache_version = abc_invalidation_counter;
+    _abcmodule_state * state = PyType_GetModuleState(type);
```

Comment:
Some compilers don't like variable definitions in the middle of a code block. Please define the variable at the top but keep the assignment here.

You need to check the result for ``NULL`` and handle the error case.